packages/docusaurus-theme-classic/src/theme/CodeBlock/index.tsx

```diff
@@ -49,13 +49,44 @@ export default function CodeBlock({
   const codeBlockTitle = parseCodeBlockTitle(metastring) || title;
   const prismTheme = usePrismTheme();
 
-  // In case interleaved Markdown (e.g. when using CodeBlock as standalone component).
+  // <pre> tags in markdown map to CodeBlocks and they may contain JSX children.
+  // When the children is not a simple string, we just return a styled block without actually highlighting.
+  if (React.Children.toArray(children).some((el) => isValidElement(el))) {
+    return (
+      <Highlight
```

Comment:
what is the purpose of using the Highlight component in this case? is it only to share styling?

Should we also add the "copy" button in such case?

Reply:
Yes, it's for styling because the background color & the font comes from the Prism renderer's theme/

I don't think it'll be very easy to add the copy button, because we just want to copy plain text, and extracting plain text from the deeply nested React children is quite tedious.